lib/Auth/Process/DatabaseCommand.php

```diff
@@ -164,8 +164,12 @@ public function insertLogin(&$request, &$date)
                 " is empty and login log wasn't inserted into the database."
             );
         } else {
-            $idAttribute = $this->databaseConnector->getUserIdAttribute();
-            $userId = isset($request['Attributes'][$idAttribute]) ? $request['Attributes'][$idAttribute][0] : null;
+            if (empty($this->databaseConnector->getUserIdAttribute())) {
+                $userIdAttribute = $request['rciamAttributes']['cuid'];
+            } else {
+                $userIdAttribute = $request['Attributes'][$this->databaseConnector->getUserIdAttribute()];
+            }
+            $userId = isset($userIdAttribute) ? $userIdAttribute[0] : null;
```

Comment:
Check if we avoid the get userid attribute logic since this is done during the module initialisation/process function

Reply:
Done